components/rich_text_editor/rich_text_editor.vue

```diff
@@ -0,0 +1,250 @@
+<template>
+  <editor-content
+    :editor="editor"
+    data-qa="dt-rich-text-editor"
+  />
+</template>
+
+<script>
+import { Editor, EditorContent } from '@tiptap/vue-2';
+import Document from '@tiptap/extension-document';
+import Paragraph from '@tiptap/extension-paragraph';
+import Text from '@tiptap/extension-text';
+import {
+  RICH_TEXT_EDITOR_OUTPUT_FORMATS,
+  RICH_TEXT_EDITOR_AUTOFOCUS_TYPES,
+} from './rich_text_editor_constants';
+
+export default {
+  name: 'DtRichTextEditor',
+
+  components: {
+    EditorContent,
+  },
+
+  props: {
+    /**
+     * Value of the input
+     */
+    value: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Whether the input is editable
+     */
+    editable: {
+      type: Boolean,
+      default: true,
+    },
+
+    /**
+     * Descriptive label for the input element
+     */
+    inputAriaLabel: {
+      type: String,
+      required: true,
+    },
```

Comment:
Do we also need to support labelled by?

Reply:
Hmm, I'm not sure. What I'm seeing is that usually the element that labels something is built into the component as a header or whatever. In this case it'd be an external element ID that would have to be passed, but I feel like it'd be easier to pass a string for the aria-label instead :?

Reply:
alright, not really a big deal it can be easily added later if needed.